src/displayapp/screens/settings/SettingSetDate.cpp

```diff
@@ -0,0 +1,272 @@
+  btnSetTime->user_data = this;
+  lv_obj_set_size(btnSetTime, 70, 40);
+  lv_obj_align(btnSetTime, lv_scr_act(), LV_ALIGN_CENTER, 0, 90);
+  lv_obj_set_style_local_value_str(btnSetTime, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, "Set");
+  lv_obj_set_event_cb(btnSetTime, event_handler);
+}
+
+SettingSetDate::~SettingSetDate() {
+  lv_obj_clean(lv_scr_act());
+}
+
+bool SettingSetDate::Refresh() {
+  return running;
+}
+
+
+void SettingSetDate::HandleButtonPress(lv_obj_t *object, lv_event_t event) {
+
+  if(object == btnDayPlus && (event == LV_EVENT_PRESSED)) {
+    dayValue++;
+    if (dayValue > MaximumDayOfMonth())
+      dayValue = 1;
+    lv_label_set_text_fmt(lblDay, "%d", dayValue);
+    lv_obj_align(lblDay, lv_scr_act(), LV_ALIGN_CENTER, POS_X_DAY, POS_Y_TEXT);
+    lv_btn_set_state(btnSetTime, LV_BTN_STATE_RELEASED);
+  }
+
+  if(object == btnDayMinus && (event == LV_EVENT_PRESSED)) {
+    dayValue--;
+    if (dayValue < 1)
+      dayValue = MaximumDayOfMonth();
+    lv_label_set_text_fmt(lblDay, "%d", dayValue);
+    lv_obj_align(lblDay, lv_scr_act(), LV_ALIGN_CENTER, POS_X_DAY, POS_Y_TEXT);
+    lv_btn_set_state(btnSetTime, LV_BTN_STATE_RELEASED);
+  }
+
+  if(object == btnMonthPlus && (event == LV_EVENT_PRESSED)) {
+    monthValue++;
+    if (monthValue > 12)
+      monthValue = 1;
+    UpdateMonthLabel();
+    lv_btn_set_state(btnSetTime, LV_BTN_STATE_RELEASED);
+    CheckDay();
+  }
+
+  if(object == btnMonthMinus && (event == LV_EVENT_PRESSED)) {
+    monthValue--;
+    if (monthValue < 1)
+      monthValue = 12;
+    UpdateMonthLabel();
+    lv_btn_set_state(btnSetTime, LV_BTN_STATE_RELEASED);
+    CheckDay();
+  }
+
+  if(object == btnYearPlus && (event == LV_EVENT_PRESSED)) {
+    yearValue++;
+    lv_label_set_text_fmt(lblYear, "%d", yearValue);
+    lv_obj_align(lblYear, lv_scr_act(), LV_ALIGN_CENTER, POS_X_YEAR, POS_Y_TEXT);
+    lv_btn_set_state(btnSetTime, LV_BTN_STATE_RELEASED);
+    CheckDay();
+  }
+
+  if(object == btnYearMinus && (event == LV_EVENT_PRESSED)) {
+    yearValue--;
+    lv_label_set_text_fmt(lblYear, "%d", yearValue);
+    lv_obj_align(lblYear, lv_scr_act(), LV_ALIGN_CENTER, POS_X_YEAR, POS_Y_TEXT);
+    lv_btn_set_state(btnSetTime, LV_BTN_STATE_RELEASED);
+    CheckDay();
+  }
+
+  if(object == btnSetTime && (event == LV_EVENT_PRESSED)) {
+    NRF_LOG_INFO("Setting date (manually) to %04d-%02d-%02d", yearValue, monthValue, dayValue);
+    dateTimeController.SetTime(static_cast<uint16_t>(yearValue),
+                               static_cast<uint8_t>(monthValue),
+                               static_cast<uint8_t>(dayValue),
+                               0,
+                               dateTimeController.Hours(),
+                               dateTimeController.Minutes(),
+                               dateTimeController.Seconds(),
+                               nrf_rtc_counter_get(portNRF_RTC_REG));
+    lv_btn_set_state(btnSetTime, LV_BTN_STATE_DISABLED);
+  }
+}
```

Comment:
Add a check to the top to reduce duplicates. CLICKED event is preferred.
```
if (event != LV_EVENT_CLICKED) {
  return;
}
```
The if statements should be turned into a switch statement.

Reply:
Added the check on the method's top.
But I'm not sure if it's possible to use a switch statement for determining the clicked button - switch is only for comparisons with literals, right?

Reply:
Oh whoops, well we can use else if instead to skip the other checks.

Reply:
Ok, I did that when I removed the superfluous event checks in the `if` statements. :)